### PR TITLE
BE-20 Leading dot for method invocation, layout cops

### PR DESCRIPTION
**Pull request by Aerdayne**
[Ticket](https://hellogetsafe.atlassian.net/browse/BE-20)

This PR changes the `DotPosition` cop to enforce leading dots when calling methods.
It also introduces a bunch of auto-correctable layout cops in order to achieve consistency of indentation across our projects.

### CODE REVIEW

**Review by alecspopa — APPROVED** [review]
👏 There were some warning on rubocop when we upgraded the version. Could you please fix those as well?
```
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file.

Please also note that you can opt-in to new cops by default by adding this to your config:
  AllCops:
    NewCops: enable

Lint/DuplicateMagicComment: # new in 1.37
  Enabled: true
Lint/UselessRescue: # new in 1.43
  Enabled: true
For more information: https://docs.rubocop.org/rubocop/versioning.html
```

**Comment by shalvah-gs** [comment]
@alecspopa they're in https://github.com/nxt-insurance/nxt_cop/pull/87

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# v1.1.0 2023-05-05
+## What's Changed
+* Change `DotPosition` to enforce leading dots for method invocation
+* Add multiple autocorrectable layout cops
+
+**Full Changelog**: https://github.com/nxt-insurance/nxt_cop/compare/v1.0.19...v1.1.0
```
[the remark this answers is not in the document]
Comment (Aerdayne) [reply CHANGELOG.md:6]: [50b6db8](https://github.com/nxt-insurance/nxt_cop/pull/88/commits/50b6db8ebd1644678538a9751641364c550d2490)

**Comment by alecspopa** [comment]
I agree with @AkihikoITOH. Better to pick a standard and go with it then to debate what is better :)

**Comment by shalvah-gs** [comment]
Please do not pick Standard. 🙏  That tool drives me crazy. 😭 

On a serious note, you know what happens when we arbitrarily pick rules? Lots of `# disable` comments. I agree that sometimes consensus is hard, but I think with a little flexibility, we can find some common ground for some things, and maybe ignore others.

Picking a random _wide-reaching_ and _inflexible_ set of rules won't make code style discussions magically stop. People will ignore the rules in some situations, and justify it somehow; we'll still end up having debates about some points; there _will_ be frustrated devs who feel forced to code a certain way for no reason other than "The tool says so". I think it's better to accept a compromise after discussion.

**Comment by alecspopa** [comment]
Sorry, my comment needs some explanation.

> Please do not pick Standard. 🙏 That tool drives me crazy. 😭

I did not mean pick `standardrb` thing and replace Rubocop. I mean pick some Rubocop config and go with it.
But I am not sure this Switzerland approach works, I think we need some North Korea approach when picking a configuration :))

**Comment by shalvah-gs** [comment]
Hmm, I don't know. I feel like it's two or three main rules we have disagreements on. We could leave those out and enable the rest. I'm willing to compromise where necessary. It's code style, not the end of the world 😀. I hate the multiline method call indentation that Aki prefers it but I can live with it (since I can replicate the other style by going to a new line).

**Comment by shalvah-gs** [comment]
> We can all hate all the rules and it would still be worth.

Right. Honestly, I wouldn't mind trying the whole "opt-in to the full ruleset and see if we all hate it but all shut up" thing as an experiment for a while. 😀 But we probably don't have the scope to run such an experiment.

**Comment by AkihikoITOH** [comment]
article-0-0A57726C000005DC-900_468x332_1_

**Review by chiibis — APPROVED** [review]

**Comment by chiibis** [comment]
> Using something like standardrb is not so much about whether we like the rules or not, but it's a decision we can make to not waste any more time on code-style debate, not now, not in the future, as if it's set in stone. We can all hate all the rules and it would still be worth. Team changes. Some "weird" styles we see in older parts of our codebase were once preferred by those who worked on it back then. There will be new members with strong opinion about code styles who want to debate them all over again. Language changes too

I double this. 

When 10+ people constantly work on the same codebase, it must be unified. It's not about personal preferences anymore.

**Comment by Aerdayne** [comment]
Merging this without several cops that proved to be controversial. Here's the result of running `rubocop -a app/` in the IS.
image

**Pull request merged**